src/pages/HomePage.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from 'react';
+import { NavLink } from 'react-router-dom';
+import Paper from '@mui/material/Paper';
+import NumberFormat from 'react-number-format';
+import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { fetchConstructors } from '../redux/constructors';
+import { fetchDrivers } from '../redux/drivers';
+import { fetchRaces } from '../redux/races';
+import { fetchSeason } from '../redux/home';
+import '../assets/css/Home.css';
+
+const Home = (props) => {
+  const [year, setYear] = useState('');
+  const { season, actualYear } = props;
+  const races = useSelector((state) => state.race);
+  const drivers = useSelector((state) => state.driver);
+  const constructors = useSelector((state) => state.constructor);
+  const dispatch = useDispatch();
+  const rgx = /(195\d|19[6-9]\d|20[01]\d|202[0-2])/;
+
+  const fetchData = () => {
+    dispatch(fetchConstructors(year));
+    dispatch(fetchDrivers(year));
+    dispatch(fetchRaces(year));
+    dispatch(fetchSeason(year));
+  };
+
+  const enterBtnHandle = (e) => {
+    if (e.key === 'Enter') {
+      if (year.match(rgx)) {
+        e.preventDefault();
+        fetchData();
+        setYear('');
+      }
+    }
+  };
+
+  const handleEvent = () => {
+    fetchData();
+    setYear('');
+  };
+
+  const changeSeason = (e) => {
+    setYear(e.target.value);
+  };
```

Comment:
- [ ]  As the previous reviewer suggested, after setting the year with setState, you should write the logic to filter the data retrieved from the API based on year 👍🏽 so that the filtered data will be rendered to the DOM 👍🏽 .